backend/data_validation/base_data_val.py

```diff
@@ -0,0 +1,228 @@
+import collections
+import json
+import statistics
+
+import falcon_alliance
+import numpy as np
+import yaml
+from utils import *
+
+
+class BaseDataValidation:
+    """
+    Base class that validates the data passed in.
+
+    Implements base checks explained below (e.g. checking if the scout scouted the right driver station.)
+    """
+
+    def __init__(self, path_to_config: str = "config.yaml"):
+        # Basic attributes
+        self.tba_match_data = {}
+        self.errors = []
+        self.match_schedule = {}
+
+        # Config attributes
+        with open(path_to_config) as file:
+            self.config = yaml.safe_load(file)
+
+        self.path_to_output_file = self.config.get("path_to_output", "errors.json")
+        self.path_to_data_file = self.config.get(
+            "path_to_data",
+            f"../data/{self.config['year']}{self.config['event_code']}_match_data.json",
+        )
+
+        # Retrieves match schedule
+        try:
+            with open(
+                self.config.get("path_to_match_schedule", "../data/match_schedule.json")
+            ) as file:
+                self.match_schedule = json.load(file)
+        except FileNotFoundError:  # We want to ignore if it doesn't exist because get_match_schedule() will create it.
+            pass
+
+        # Setting up FalconAlliance (our connection to TBA) and retrieving match schedule if it doesn't exist
+        self.api_client = falcon_alliance.ApiClient(
+            api_key="6lcmneN5bBDYpC47FolBxp2RZa4AbQCVpmKMSKw9x9btKt7da5yMzVamJYk0XDBm"  # for testing purposes
```

Comment:
Not sure how strict we want to be with this. Usually you don't want to expose these to public so it's recommended to put it in your environment variables and read from there.

Reply:
github env variables are funky,, worth figuring out for the future but for strictly testing purposes it's fine imo. tba api keys are the only api keys i would recommend exposing at all (even then shouldn't expose once we release the app). a lot of teams expose tba api keys bc it's a lot of effort to hide as well.